resources/workflow-scripts/trigger-graalvm-check.bal

```diff
@@ -0,0 +1,291 @@
+                }
+            }
+        } else {
+            log:printInfo("GraalVM check is in progress", module = module_name, link = moduleStatus.link);
+        }
+    } on fail error err {
+        log:printError("Failed to get the GraalVM check status", module = module_name, 'error = err);
+    }
+}
+
+function getWorkflowJobStatuses(string module_name, int workflow_id) returns JobRunStatus[]|error {
+    record {JobRunStatus[] jobs;} jobRunStatuses = check gitHubClient->/[ORG_NAME]/[module_name]/actions/runs/[workflow_id]/jobs(headers);
+    return jobRunStatuses.jobs;
+}
+
+function isComplete(map<LevelStatus> result) returns boolean {
+    foreach LevelStatus levelStatus in result {
+        if levelStatus.status != COMPLETED {
+            return false;
+        }
+    }
+    return true;
+}
+
+function isLevelComplete(LevelStatus levelStatus) returns boolean {
+    foreach map<ModuleStatus> moduleStatusMap in levelStatus.modules {
+        foreach [string, ModuleStatus] [_, moduleStatus] in moduleStatusMap.entries() {
+            if moduleStatus.status != COMPLETED {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+type ReportRecord record {|
+    string level;
+    string module;
+    string ubuntuStatus = NOT_APPLICABLE;
+    string windowsStatus = NOT_APPLICABLE;
+|};
+
+function createReport(map<LevelStatus> result) returns error? {
+    table<ReportRecord> resultTable = table [];
+    foreach [string, LevelStatus] [level, levelStatus] in result.entries() {
+        foreach map<ModuleStatus> moduleStatusMap in levelStatus.modules {
+            foreach [string, ModuleStatus] [module, moduleStatus] in moduleStatusMap.entries() {
+                ReportRecord rec = {
+                    level: "Level " + level,
+                    module: module,
+                    ubuntuStatus: string `[${moduleStatus.jobs.ubuntu?.conclusion ?: NOT_APPLICABLE}](${moduleStatus.jobs?.ubuntu?.link ?: moduleStatus.link})`,
+                    windowsStatus: string `[${moduleStatus.jobs.windows?.conclusion ?: NOT_APPLICABLE}](${moduleStatus.jobs?.windows?.link ?: moduleStatus.link})`
+                };
+                resultTable.add(rec);
+            }
+        }
+    }
+
+    string title = "GraalVM Check Report";
+    string tableTitle = "| Level | Module | Ubuntu Build | Windows Build |";
+    string tableTitleSeparator = "| ----- | ------ | :----: | :----: |";
+    string[] rows = from ReportRecord reportRecord in resultTable
+        select
+        "| " + string:'join(" | ", reportRecord.level, reportRecord.module, reportRecord.ubuntuStatus, reportRecord.windowsStatus) + " |";
+    string[] summary = ["## " + title + " :rocket:", tableTitle, tableTitleSeparator, ...rows];
+    string summaryString = string:'join("\n", ...summary);
+    check io:fileWriteString("graalvm_check_summary.md", summaryString);
```

Comment:
Can't we add this as a separate section in the dashboard?

Reply:
We are already showing the daily GraalVM checks in the dashboard in a separate column. So I don't think there is a requirement to show this in the dashboard.